lib/Dialect/Torch/Transforms/ScalarizeShapes.cpp

```diff
@@ -99,14 +82,62 @@ LogicalResult getListFromTensor(Value value, SmallVector<Value> &vals) {
     if (ty.getSizes()[0] > kMaxFold)
       return failure();
 
-    vals.resize(vals.size() + ty.getSizes()[0], full.getFillValue());
+    vals.resize(vals.size() + ty.getSizes()[0],
+                getAsOpFoldResult(full.getFillValue()));
     return success();
   }
+  // TODO: Add a case for unsqueeze of a primnumtotensorscalarop?
+
+  // Last supported case: ValueTensorLiteralOp
+  auto literalOp = value.getDefiningOp<Torch::ValueTensorLiteralOp>();
+  if (!literalOp)
+    return failure();
+
+  // Check the type. We make sure the type is not unsigned here before trying to
+  // materialize
+  auto ty = cast<ValueTensorType>(literalOp.getType());
+  if (!ty.hasSizes() || ty.getSizes().size() > 1)
+    return failure();
+  int64_t listSize = ty.getSizes().size() == 1 ? ty.getSizes().front() : 1;
+  auto intTy = dyn_cast_or_null<IntegerType>(ty.getDtype());
+  if (!intTy || intTy.isUnsigned())
+    return failure();
+
+  if (auto splattr =
```

Comment:
You can probably restructure this a bit for clarity:

```
auto splattr = dyn_cast_or_null<SplatElementsAttr>(literalOp.getValue());
auto denseAttr = dyn_cast_or_null<DenseIntElementsAttr>(literalOp.getValue());
if (!splattr && !denseAttr) return false;
if (splattr) ...
if (denseAttr) ...
```

It just handles the escape case first and makes it clear what the block is trying to express